python/mlcroissant/mlcroissant/_src/operation_graph/operations/join.py

```diff
@@ -27,7 +27,7 @@ def __call__(self, *args: pd.Series) -> pd.Series | None:
             raise ValueError(f"Unsupported: Trying to join {len(args)} pd.DataFrames.")
         fields = self.node.fields
         # `joins` is the list of joins to execute (source1, df1) x (source2, df2).
-        joins: list[tuple[tuple[Source, pd.DataFrame], tuple[Source, pd.DataFrame]]] = (
+        joins: list[tuple[tuple[Source, pd.Series], tuple[Source, pd.Series]]] = (
```

Comment:
Can you please configure pytype to track those errors on GitHub? Otherwise, we'll always run behind those errors.

I found out that changing pytype parameters in [`./python/mlcroissant/pyproject.toml#L56`](https://github.com/mlcommons/croissant/blob/main/python/mlcroissant/pyproject.toml#L56) has an impact.

An example of another failing type checks: in [`operation_graph/operations/read.py#L82`](https://github.com/mlcommons/croissant/blob/main/python/mlcroissant/mlcroissant/_src/operation_graph/operations/read.py#L82): `content_url` could be None.

Reply:
Addressed in #265 and following